src/partition/repart.c

```diff
@@ -6807,6 +6840,35 @@ static int parse_argv(int argc, char *argv[]) {
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Expected at most one argument, the path to the block device.");
 
+        if (arg_make_ddi) {
+                if (arg_definitions)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Combination of --make-ddi= and --definitions= is not supported.");
+                if (!IN_SET(arg_empty, EMPTY_UNSET, EMPTY_CREATE))
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Combination of --make-ddi= and --empty= is not supported.");
+
+                /* Imply automatic sizing in DDI mode */
+                if (arg_size == UINT64_MAX && !arg_size_auto)
+                        arg_size_auto = true;
```

Comment:
I'd just drop the `arg_size_auto` check. If it's already `true` it doesn't hurt to set it `true` again.